cmake/modules/FindLEX.cmake

```diff
@@ -0,0 +1,73 @@
+find_program(LEX_EXECUTABLE NAMES lex DOC "path to the lex executable")
```

Comment:
Flex does not exist on z/OS so it is not an option. Lex is POSIX so it is available on z/OS as well as all other platforms OMR supports.

Reply:
I meant more along the line of tricking/forcing cmake into accepting lex as  flex so that we can avoid having a duplicate `[F]LEX_TARGET()`

Reply:
This is not possible unfortunately as `lex` and `flex` are not command line option compatible.

Reply:
Also I don't think we have duplicate targets. The commits here merely migrate OMR to use `lex` and `yacc` instead of `flex` and `bison` since the former are supported on all POSIX systems. There should be no usage of the latter following this change.

In fact I think we can likely contribute FindLEX.cmake and FindYACC.cmake to upstream CMake project.